sqldelight-compiler/dialect/src/main/kotlin/app/cash/sqldelight/dialect/api/ReturningQueryable.kt

```diff
@@ -0,0 +1,37 @@
+package app.cash.sqldelight.dialect.api
+
+import com.alecstrong.sql.psi.core.psi.QueryElement
+import com.alecstrong.sql.psi.core.psi.Queryable
+import com.alecstrong.sql.psi.core.psi.SqlAnnotatedElement
+import com.alecstrong.sql.psi.core.psi.SqlTableName
+import com.intellij.psi.util.PsiTreeUtil
+
+/**
+ * Query deriving from the `RETURNING` clause of an expression.
+ *
+ * Typical use cases include `INSERT`, `UPDATE`, and `DELETE`. This class is similar to [SelectQueryable] but differs in
+ * the fact that only 1 table can be part of the query, and that table is guaranteed to be "real".
+ *
+ * @param statement Parent statement. Typically, this is the `INSERT`, `UPDATE`, or `DELETE` statement.
+ * @param select The `RETURNING` clause of the statement. Represented as a query since it returns values to the caller.
+ * @param tableName Name of the table the [statement] is operating on.
+ */
+class ReturningQueryable(
```

Comment:
Used by SQLite and Postgres dialects at the moment, but reasonable to think additional dialects would want this

Reply:
Exactly, thanks.